File: src/test/java/no/nav/samordning/innlasting/SamordningHendelseMapperTest.java
package no.nav.samordning.innlasting;

import no.nav.samordning.schema.SamordningHendelse;
import org.junit.jupiter.api.Test;
import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;


import static org.junit.jupiter.api.Assertions.assertEquals;

class SamordningHendelseMapperTest {

    private static final String VEDTAK_ID = "JKL678";
    private static final String IDENTIFIKATOR = "987654321";
    private static final String YTELSES_TYPE = "AAP";
    private static final String FOM = "2000-01-01";
    private static final String TOM = "2010-02-04";

    @Test
    void mapToJson() throws Exception {

        SamordningHendelse samordningHendelse = new SamordningHendelse();
        samordningHendelse.setVedtakId(VEDTAK_ID);
        samordningHendelse.setIdentifikator(IDENTIFIKATOR);
        samordningHendelse.setYtelsesType(YTELSES_TYPE);
        samordningHendelse.setFom(FOM);
        samordningHendelse.setTom(TOM);

        String expectedHendelse = "{" +
                "\"identifikator\": \"" + IDENTIFIKATOR + "\", " +
                "\"ytelsesType\": \"" + YTELSES_TYPE + "\", " +
                "\"vedtakId\": \"" + VEDTAK_ID + "\", " +
                "\"fom\": \"" + FOM + "\", " +
                "\"tom\": \"" + TOM + "\"}";
        String expectedHendelseJson = new ObjectMapper().writeValueAsString(expectedHendelse);

        assertEquals(expectedHendelseJson, SamordningHendelseMapper.toJson(samordningHendelse));
    }
}

File: src/test/kotlin/no/nav/samordning/innlasting/SamordningHendelseMapperTest.kt
package no.nav.samordning.innlasting


import com.fasterxml.jackson.databind.ObjectMapper
import no.nav.samordning.schema.SamordningHendelse
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class SamordningHendelseMapperTest {

    @Test
    @Throws(Exception::class)
    fun mapToJson() {

        val samordningHendelse = SamordningHendelse().apply {
            setVedtakId(VEDTAK_ID)
            setIdentifikator(IDENTIFIKATOR)
            setYtelsesType(YTELSES_TYPE)
            setSamId(SAM_ID)
            setFom(FOM)
            setTom(TOM)
        }

        val expectedHendelse = """{"identifikator": "$IDENTIFIKATOR", "ytelsesType": "$YTELSES_TYPE", "vedtakId": "$VEDTAK_ID", "samId": "$SAM_ID", "fom": "$FOM", "tom": "$TOM"}"""
        val expectedHendelseJson = ObjectMapper().writeValueAsString(expectedHendelse)

        assertEquals(expectedHendelseJson, SamordningHendelseMapper.toJson(samordningHendelse))
    }

    companion object {

        private const val VEDTAK_ID = "JKL678"
        private const val SAM_ID = "BOGUS"
        private const val IDENTIFIKATOR = "987654321"
        private const val YTELSES_TYPE = "AAP"
        private const val FOM = "2000-01-01"
        private const val TOM = "2010-02-04"
    }
}
